src/engine.js

```diff
@@ -0,0 +1,187 @@
+var _ = require('underscore');
+var AnalysisPoller = require('./analysis/analysis-poller');
+var AnonymousMapSerializer = require('./windshaft/map-serializer/anonymous-map-serializer/anonymous-map-serializer');
+var Backbone = require('backbone');
+var CartoDBLayerGroup = require('./geo/cartodb-layer-group');
+var DataviewsCollection = require('./dataviews/dataviews-collection');
+var LayersCollection = require('./geo/map/layers');
+var ModelUpdater = require('./windshaft-integration/model-updater');
+var NamedMapSerializer = require('./windshaft/map-serializer/named-map-serializer/named-map-serializer');
+var Request = require('./windshaft/request');
+var Response = require('./windshaft/response');
+var WindshaftClient = require('./windshaft/client');
+
+/**
+ *
+ * Creates a new Engine.
+ * An engine is the core of a carto app.
+ *
+ * With the help of external services the engine will:
+ *  - Keep the state of the layers and dataviews.
+ *  - Serialize the state and send requests to the server.
+ *  - Parse the server response and update the internal models.
+ *  - Notify errors or successful operations.
+ * 
+ * @param {Object} params - The parameters to initialize the engine.
+ * @param {string} params.apiKey - Api key used to be autenticate in the windshaft server.
+ * @param {string} params.authToken - Token used to be autenticate in the windshaft server.
+ * @param {string} params.username - Name of the user registered in the windshaft server.
+ * @param {string} params.serverUrl - Url of the windshaft server.
+ * @param {boolean} params.templateName - While we dont remove named maps we must explicitly say when the map is named. Defaults to false.
+ * @param {boolean} params.statTag - Token used to get map view statistics.
+ * @class
+ */
+function Engine (params) {
+  if (!params) throw new Error('new Engine() called with no paramters');
+  this._apiKey = params.apiKey;
+  this._authToken = params.authToken;
+  this._username = params.username;
+  this._serverUrl = params.serverUrl;
```

Comment:
Some of these instance variables (eg: `this._serverUrl`) are only being used in this context (when we declare `windshaftSettings`). I wouldn't personally use instance variables if they are not used elsewhere.